src/models/device.ts

```diff
@@ -261,21 +261,50 @@ const getDeviceModel = function (
 	};
 
 	const set = async (
-		uuidOrIdOrArray: string | string[] | number | number[],
+		filterOrUuidOrIdOrArray:
+			| (PineFilter<Device> & { belongs_to__application: number })
+			| string
+			| string[]
+			| number
+			| number[],
 		body: SubmitBody<Device>,
 	): Promise<void> => {
+		if (
+			typeof filterOrUuidOrIdOrArray === 'object' &&
+			!Array.isArray(filterOrUuidOrIdOrArray) &&
+			!('belongs_to__application' in filterOrUuidOrIdOrArray)
+		) {
+			throw new Error('Filter must contain `belongs_to__application`.');
+		}
+
+		const patchRequest = async (
+			filter: PineFilter<Device>,
+			requestBody: SubmitBody<Device>,
+		) => {
+			await pine.patch<Device>({
+				resource: 'device',
+				options: {
+					$filter: filter,
+				},
+				body: requestBody,
+			});
+		};
+		if (
+			typeof filterOrUuidOrIdOrArray === 'object' &&
+			'belongs_to__application' in filterOrUuidOrIdOrArray
+		) {
```

Comment:
How about moving the parameter check in here?
```suggestion
		if (
			!Array.isArray(filterOrUuidOrIdOrArray) &&
			typeof filterOrUuidOrIdOrArray === 'object'
		) {
			if (!('belongs_to__application' in filterOrUuidOrIdOrArray) || typeof filterOrUuidOrIdOrArray.belongs_to__application !== 'number') {
				throw new InvalidParameterError('Filter must contain a numeric `belongs_to__application` property.');
			}
			// ... do the patch
```